java/cdk-custom-resources/src/test/java/sleeper/cdk/custom/TableDefinerLambdaIT.java

```diff
@@ -153,6 +156,49 @@ public void shouldCreateTableWithMultipleSplitPoints() throws IOException {
                             .splitToNewChildren("r", "rl", "rr", Long.valueOf(10))
                             .buildList());
         }
+
+        @Test
+        public void shouldFailToReuseTableIfTableDoesNotExist() throws IOException {
+            //Given
+            tableProperties.set(REUSE_EXISTING_TABLE, "true");
+
+            //Then
+            assertThatThrownBy(() -> callLambda(CREATE, tableProperties))
+                    .isInstanceOf(NoTableToReuseException.class);
+        }
+
+        @Test
+        public void shouldFailToReuseTableIfImportFlagNotSet() throws IOException {
+            //Given
+            callLambda(CREATE, tableProperties);
+            //Should just take the table offline
+            tableProperties.set(RETAIN_DATA_AFTER_DELETE, "true");
+            callLambda(DELETE, tableProperties);
+
+            //Then
+            assertThatThrownBy(() -> callLambda(CREATE, tableProperties))
+                    .isInstanceOf(TableAlreadyExistsException.class)
+                    .hasMessageContaining("If attempting to reuse an existing table ensure the sleeper.reuse.existing.table property is set to true.");
```

Comment:
If we set the table ID and name in the test, I think we could assert on the full message?